File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var longestConsecutive = function(nums) {
    if(nums.length === 0) return 0
    const set = new Set()
    for(const num of nums) {
        set.add(num)
    }
    let maxCount = 1
    for(const num of nums) {
        if(!set.has(num-1)) {
            let curr = num
            while(set.has(curr+1)) {
                curr+=1
            }
            maxCount = Math.max(maxCount, curr-num+1)
        }
    }
    return maxCount
};

File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.java
class Solution {
    public int longestConsecutive(int[] nums) {
        Set<Integer> set = new HashSet();
        for(int num: nums) {
            set.add(num);
        }
        int res = 0;
        for(int num: nums) {
            if(!set.contains(num-1)) {
                int curr = num;
                while(set.contains(curr+1)) {
                    curr += 1;
                }
                res = Math.max(res, curr-num+1);
            }
        }
        return res;
    }
}